### DISCUSSION TITLE
Upgraded operator-sdk to 1.35.0 and migrated kubebuilder go/v3 to go/v4

### DISCUSSION THREAD

**Discussion by iam-veeramalla** [argoproj-labs/argocd-operator#1472] [Show and tell]
Shout out to @saumeya for upgrading the operator-sdk to v1.35.0 and also migrating the kubebuilder from go/v3 to go/v4.

Also, thanks to @anandrkskd for working along with @saumeya on this.

This change will be part of the next minor release.

**Comment by saumeya** [comment]
Thanks @iam-veeramalla 
Thanks @svghadi for the reviews and feedback!
